src/components/QuranReader/utils/memoization.ts

```diff
@@ -30,7 +32,7 @@ export const verseFontChanged = (
   prevWords: Word[],
   nextWords: Word[],
 ): boolean =>
-  prevQuranStyles.mushafLines !== nextQuranStyles.mushafLines ||
+  !isEqual(prevQuranStyles, nextQuranStyles) ||
```

Comment:
using `isEqual` is a great solution but I have 2 comments:

1. We try to stay away from using packages as much as possible if the same can be easily achieved manually.
2. Comparing the entire `quranStyles` object would result in un-necessary re-renders even if the changes are not related to the translation view e.g. changing the Tafsir font while the user is viewing translation will cause every word in the translation view to re-render. You can check the profiler snapshot below for the case of changing the Tafsir font, while viewing the translations, all verses re-rendered:

<img width="1061" alt="Screen Shot 2021-11-21 at 09 00 07" src="https://user-images.githubusercontent.com/15169499/142746492-2f6b7962-e495-49aa-983c-b807bed0560d.png">

so I suggest targeting only the translation font `prev` and `next` and doing the comparison inside `TranslationViewCell.areVersesEqual` because the translation font change is only related to the translation view and not the reading view. JazakAllah Khairan

Reply:
@osamasayed Jazakallahu khair for review the PR, I have update the PR as suggested. I have added a separate helper `verseTranslationFontChanged` similar to `verseFontChanged` for comparing the translation font style changes. Do let me know WDYT? 

Shukran

Reply:
Looks great to me and good to merge! thanks a lot!